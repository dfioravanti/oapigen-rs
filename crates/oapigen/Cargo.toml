[package]
name = "oapigen"
description = "foo"
version = "0.0.1"

categories = ["data-structures", "parsing"]
repository = { workspace = true }
license = { workspace = true }
edition = { workspace = true }
rust-version = { workspace = true }

[package.metadata.docs.rs]
all-features = true

[dependencies]
oas3 = { workspace = true, features = ["yaml-spec"] }
serde_json = { workspace = true }
serde_yaml = { workspace = true }
serde = { workspace = true, features = ["derive"] }
quote = { workspace = true }
proc-macro2 = { workspace = true }
prettyplease = { workspace = true }
syn = { workspace = true }
log = { workspace = true }
indoc = { workspace = true }
thiserror = { workspace = true }
rstest = { workspace = true }
convert_case = { workspace = true }
clap = { workspace = true, features = ["derive"] }
jiff = "0.2.15"

[dev-dependencies]
insta = { workspace = true, features = ["yaml"] }
