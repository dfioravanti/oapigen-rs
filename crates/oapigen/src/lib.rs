pub mod generating;
mod models;
pub mod parsing;

pub(crate) mod format;

pub use generating::spec_to_rust;
pub use models::*;
