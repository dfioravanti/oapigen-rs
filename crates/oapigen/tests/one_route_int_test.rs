use insta;
use oapigen;

#[test]
fn one_route_int_test() {
    let config = oapigen::Config::default();
    let f = std::fs::File::open("fixtures/one_route_int.yaml").unwrap();
    let spec: oas3::Spec = serde_yaml::from_reader(f).unwrap();

    let got = oapigen::spec_to_rust(&config, spec).unwrap();

    insta::assert_snapshot!(got.to_string());
}
