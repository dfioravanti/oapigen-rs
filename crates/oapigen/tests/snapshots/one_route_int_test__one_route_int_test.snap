---
source: crates/oapigen/tests/one_route_int_test.rs
expression: got.to_string()
---
struct ListAllWidgetsResponse200(i32);
